packages/@aws-cdk/aws-servicediscovery/lib/service.ts

```diff
@@ -0,0 +1,448 @@
+ */
+export class Service extends cdk.Construct implements IService {
+  /**
+   * A name for the Cloudmap Service.
+   */
+  public readonly serviceName: string;
+
+  /**
+   *  The namespace for the Cloudmap Service.
+   */
+  public readonly namespace: INamespace;
+
+  /**
+   * The ID of the namespace that you want to use for DNS configuration.
+   */
+  public readonly serviceId: string;
+
+  /**
+   * The Arn of the namespace that you want to use for DNS configuration.
+   */
+  public readonly serviceArn: string;
+
+  /**
+   * The DnsRecordType used by the service
+   */
+  public readonly dnsRecordType: DnsRecordType;
+
+  /**
+   * The Routing Policy used by the service
+   */
+  public readonly routingPolicy: RoutingPolicy;
+
+  // FIXME make this only called through #createService on namespace classes?
+  constructor(scope: cdk.Construct, id: string, props: ServiceProps) {
+    super(scope, id);
+
+    const namespaceType = props.namespace.type;
+
+    // Validations
+    if (namespaceType === NamespaceType.Http && (props.routingPolicy || props.dnsRecordType)) {
+      throw new Error('Cannot specify `routingPolicy` or `dnsRecord` when using an HTTP namespace.');
+    }
+
+    if (props.healthCheckConfig && props.healthCheckCustomConfig) {
+      throw new Error('Cannot specify both `healthCheckConfig` and `healthCheckCustomConfig`.');
+    }
+
+    if (namespaceType !== NamespaceType.DnsPublic && props.healthCheckConfig) {
+      throw new Error('Can only use `healthCheckConfig` for a Public DNS namespace.');
+    }
+
+    if (props.routingPolicy === RoutingPolicy.Multivalue
+        && props.dnsRecordType === DnsRecordType.Cname) {
+      throw new Error('Cannot use `CNAME` record when routing policy is `Multivalue`.');
+    }
+
+    if (props.routingPolicy === RoutingPolicy.Multivalue
+        && props.loadBalancer) {
+      throw new Error('Cannot register loadbalancers when routing policy is `Multivalue`.');
+    }
+
+    if (props.healthCheckConfig
+        && props.healthCheckConfig.type === HealthCheckType.Tcp
+        && props.healthCheckConfig.resourcePath) {
+          throw new Error('Cannot specify `resourcePath` when using a `TCP` health check.');
+    }
+
+    // Set defaults where necessary
+    const routingPolicy = (props.dnsRecordType === DnsRecordType.Cname) || props.loadBalancer
+      ? RoutingPolicy.Weighted
+      : RoutingPolicy.Multivalue;
+
+    const dnsRecordType = props.dnsRecordType !== undefined ? props.dnsRecordType : DnsRecordType.A;
+
+    if (props.loadBalancer
+      && (!(dnsRecordType === DnsRecordType.A
+        || dnsRecordType === DnsRecordType.AAAA
+        || dnsRecordType === DnsRecordType.A_AAAA))) {
+      throw new Error('Must support `A` or `AAAA` records to register loadbalancers.');
+    }
+
+    const dnsConfig = props.namespace.type === NamespaceType.Http
+      ? undefined
+      : {
+          dnsRecords: _getDnsRecords(dnsRecordType, props.dnsTtlSec),
```

Comment:
We usually call those `render<something>` not `get<something>`. Not a big deal though.